tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,246 @@
+import os
+import pytest
+from rancher import ApiError
+from .common import *  # NOQA
+from .test_rke_cluster_provisioning import rke_config
+from .test_rke_cluster_provisioning import random_node_name
+from .test_rke_cluster_provisioning import create_and_validate_cluster
+
+DO_ACCESSKEY = os.environ.get('DO_ACCESSKEY', "None")
+engine_install_url = "https://releases.rancher.com/install-docker/19.03.sh"
+client = None
+admin_client = None
+user1_client = None
+
+# --------------------- rbac tests for node template -----------------------
+
+@if_test_rbac
+def test_rbac_node_template_create(remove_resource):
+    # As std user, create a node template
+    node_template = create_node_template_do(client)
+    templates = client.list_node_template(name=node_template.name)
+    assert len(templates) == 1
+    remove_resource(node_template)
```

Comment:
`remove_resource(node_template)` is apt when added right after the resource creation. For instance, if the assert fails here, the remove_resource will not get executed, and the node template will not be deleted. (for all the `remove_resource` usages)

Reply:
Addressed in changes. `remove_resource` is added right after resource creation.